examples/server/CMakeLists.txt

```diff
@@ -1,7 +1,26 @@
+set(SERVER_ASSETS
+    index.html
+    index.js
+    completion.js
+    json-schema-to-grammar.mjs
+)
+foreach(asset ${SERVER_ASSETS})
+    # Portable equivalent of `cd public && xxd -i ${asset} ../${asset}.hpp`
+    set(input "${CMAKE_CURRENT_SOURCE_DIR}/public/${asset}")
+    set(output "${CMAKE_CURRENT_BINARY_DIR}/${asset}.hpp")
+
+    string(REGEX REPLACE "\\.| |-" "_" name "${asset}")
+    file(READ "${input}" data HEX)
+    string(LENGTH ${data} hex_len)
+    math(EXPR len "${hex_len} / 2")
+    string(REGEX REPLACE "([0-9a-f][0-9a-f])" "0x\\1," data ${data})
+    file(WRITE "${output}" "unsigned char ${name}[] = {${data}};\nunsigned int ${name}_len = ${len};\n")
+endforeach()
```

Comment:
Doing this at configure time seems wrong - if the files change between running `cmake` (which shouldn't need to be done frequently) and running `make` or `cmake --build`, they will not be updated.